src/main/java/com/danielfrak/code/keycloak/providers/rest/rest/HttpClient.java

```diff
@@ -0,0 +1,10 @@
+package com.danielfrak.code.keycloak.providers.rest.rest;
+
+import java.io.IOException;
+
+public interface HttpClient {
+    void enableBasicAuth(String basicAuthUser, String basicAuthPassword);
+    void enableBearerTokenAuth(String token);
+    Response get(String uri) throws IOException;
```

Comment:
`throws IOException` seems to leak an implementation detail. Wouldn't it make more sense to rethrow a custom `HttpClientExeption extends RuntimeException`  inside of the methods, instead? That way, we wouldn't have to handle the exception in the `RestUserService`. This would then be logged by Keycloak itself, saving us the hassle.